pkg/kcp/provider/azure/redisinstance/createPrivateEndPoint.go

```diff
@@ -18,7 +18,7 @@ import (
 func createPrivateEndPoint(ctx context.Context, st composed.State) (error, context.Context) {
 	state := st.(*State)
 	logger := composed.LoggerFromCtx(ctx)
-	if state.privateEndPoint != nil {
+	if state.privateEndPoint != nil && ptr.Deref(state.privateEndPoint.Properties.ProvisioningState, "") != armnetwork.ProvisioningStateFailed {
```

Comment:
```go
const (
	ProvisioningStateDeleting  ProvisioningState = "Deleting"
	ProvisioningStateFailed    ProvisioningState = "Failed"
	ProvisioningStateSucceeded ProvisioningState = "Succeeded"
	ProvisioningStateUpdating  ProvisioningState = "Updating"
)
```
What if it's in some transient state like `Updating` or `Deleting`? I understand it might be an edge case, but just make it be able to cleanup/deprovision/delete itself and all created cloud resources. Just don't make it stuck

Reply:
If the privateEndPoint is present, any other state than "Failed" should terminate the function execution. 

`Updating` or `Deleting` means we do not want to `BeginCreateOrUpdate` the resource.

Reply:
By "termiinate" I guess you mean return. So, what will happen if it returns and the flow continues, but PEP is Updating or Deleting?

Reply:
`Updating` and `Deleting` is now handled by `waitPrivateEndPointAvailable` action which will not let the workflow to move forward unless PEP is in `Success` provisioning state.